common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -369,24 +364,17 @@ public class @K@ObjectHashMap<V> implements @K@ObjectMap<V>, Iterable<@K@ObjectM
         size++;
 
         if (size > maxSize) {
-            // Need to grow the arrays. We take care to detect integer overflow,
-            // also limit array size to ArrayList.MAX_ARRAY_SIZE.
-            rehash(adjustCapacity((int) Math.min(keys.length * 2.0, Integer.MAX_VALUE - 8)));
-        } else if (size == keys.length) {
-            // Open addressing requires that we have at least 1 slot available. Need to refresh
-            // the arrays to clear any removed elements.
-            rehash(keys.length);
+            // Double the capacity.
+            int newCapacity = keys.length << 1;
+            if(newCapacity < 0) {
```

Comment:
nit: I think this roll over behavior is well defined but should we just check if capacity would be exceeded before doing the bit shift?  We could then inline `rehash(keys.length << 1)`.